_source/_posts/2020-06-01-node-postgres-weight-tracker.md

```diff
@@ -0,0 +1,1282 @@
+
+```js
+"use strict";
+
+const addMeasurements = {
+  method: "GET",
+  path: "/add",
+  handler: ( request, h ) => {
+    return h.view( "add", { title: "Add Measurement" } );
+  }
+};
+
+const listMeasurements = {
+  method: "GET",
+  path: "/list",
+  handler: ( request, h ) => {
+    return h.view( "list", { title: "Measurements" } );
+  }
+};
+
+module.exports = [
+  addMeasurements,
+  listMeasurements
+];
+```
+
+Now update the `src/routes/index.js` file to include the `api` and `measurements` modules.
+
+```js
+"use strict";
+
+const path = require( "path" );
+
+const auth = require( "./auth" );
+const api = require( "./api" );
+const measurements = require( "./measurements" );
+
+const home = {
+  method: "GET",
+  path: "/",
+  options: {
+    auth: {
+      mode: "try"
+    },
+    handler: ( request, h ) => {
+      return h.view( "index", { title: "Home" } );
+    }
+  }
+};
+
+const staticAssets = {
+  method: "GET",
+  path: "/assets/{param*}",
+  handler: {
+    directory:{ 
+      path: path.join( __dirname, "..", "assets" )
+    }
+  },
+  options: { auth: false }
+};
+
+const error404 = {
+  method: "*",
+  path: "/{any*}",
+  handler: function ( request, h ) {
+    return h.view( "404", { title: "Not Found" } ).code( 404 );
+  },
+  options: { auth: false }
+};
+
+module.exports = [
+  home,
+  staticAssets,
+  error404
+].concat( api, auth, measurements );
+```
+
+You are now ready to test your weight tracker app! You should be able to add measurements and track your progress.
+
+## Learn More About Building Secure Node.js Apps
+
+The entire source code for this project is available at [GitHub](https://github.com/oktadeveloper/okta-nodejs-postgres-weight-tracker-example). I believe this project would be a great foundation for many types of Web applications built using Node.js.
+
+> Note: When deploying the application to a production environment, you must create a new `.env` file or use real environment variables to configure the application. Values such as the PostgreSQL connection information, `HOST_URL`, `COOKIE_ENCRYPT_PWD`, and `NODE_ENV` configuration _must_ be updated to reflect the new environment.
+
+If you liked this post, you may also enjoy some of my other posts related to Node.js and security on the Okta developer blog!
+
+* [Use TypeScript to Build a Node API with Express](/blog/2018/11/15/node-express-typescript)
+* [Build a Secure Node.js App with SQL Server](/blog/2019/03/11/node-sql-server)
+* [Add Authentication and Personalization to VuePress](/blog/2019/06/14/add-authentication-and-personalization-to-vuepress)
+* [Build a Command Line Application with Node.js](/blog/2019/06/18/command-line-app-with-nodejs)
+* [Top 10 Visual Studio Code Extensions for Node.js](/blog/2019/05/08/top-vs-code-extensions-for-nodejs-developers)
+* [An Illustrated Guide to OAuth and OpenID Connect](/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc)
```

Comment:
Add calls to action for following @oktadev and subscribing to our YouTube channel.

Reply:
Added!